src/compiler/crystal/exception.cr

```diff
@@ -140,10 +140,15 @@ module Crystal
         decorator = line_number_decorator(line_number)
         lstripped_line = line.lstrip
         space_delta = line.chars.size - lstripped_line.chars.size
+        final_column_number = if column_number < space_delta
+                                1
+                              else
+                                column_number - space_delta
+                              end
```

Comment:
Could make use of `clamp` for this:
```suggestion
        final_column_number = (column_number - space_delta).clamp(0..)
```

I think `0` is a better indicator for obviously bogus column number than `1` because it's a clear distinction.

Reply:
I chose `1` because it's how the column number is computed, as I could see when I added some puts to see a bit how things were working.

Anyway, I didn't know about `clamp`, this is way cleaner like this.

Reply:
@straight-shoota So what your thoughts about this ? Should I fix the code with `final_column_number = (column_number - space_delta).clamp(1..)` instead ? If we stick on `0` I think it would impact the way the code currently works.

Reply:
I'd say 0 is fine. But let's wait for other's input on this.

Reply:
`0` to me is a clear indicator that sth went wrong - there's a check for that in raven.cr for instance, so 👍

Reply:
I'm probably misunderstanding something because I can't understand why we need to use `0`. We want it to work as expected, so to be `1` at least. Using `0` is not how it seems to work now, so by using `0` we will change the behavior.

If we need to add debug to know what is going wrong for values that are below `1` we could send an arbitrary exception to Sentry.

Reply:
IMO `0` should only be used in case when we know that the calculations are off. Otherwise it should obviously use `1` as the beginning of the line.

Reply:
But what is the expected behavior by using this? It seems it won't throw any error at all and stack trace display will be just off by one character, isn't it?

Reply:
Stack trace is off anyways, isn't it? Using `0` just makes it clear that it's not a correct value (`1` could be an actual column number).

Reply:
Not really convinced but let's go for it!